api/v1beta1/condition_consts.go

```diff
@@ -132,9 +132,9 @@ const (
 )
 
 const (
-	// MachineHealthCheckSuccededCondition is set on machines that have passed a healthcheck by the MachineHealthCheck controller.
+	// MachineHealthCheckSucceededCondition is set on machines that have passed a healthcheck by the MachineHealthCheck controller.
```

Comment:
I guess all this is technically a breaking change and so should be treated as such.
We need to deprecate the existing variable before fully dropping it and mark the PR as breaking.

Reply:
Thanks for pointing that out I missed that this would be breaking.

Reply:
Sorry. I am new here. What's the
`Breaking` meaning? And what should I do then?

Reply:
Should I spilt this pr for it?

Reply:
hey @qmloong thanks and welcome! Because the variable being changed is exported, this change would break potential consumers during an upgrade and so this is considered a breaking change.
For breaking changes to be less disruptable there's a policy we need be compliant with https://github.com/kubernetes-sigs/cluster-api/blob/main/CONTRIBUTING.md#versioning.
Basically we need to keep backward compatibility temporarily by marking the variable as deprecated instead of removing it right away. See https://github.com/kubernetes-sigs/cluster-api/pull/5545 as a similar example.
Also this PR needs to me marked as breaking change with warning label as indicated in https://github.com/kubernetes-sigs/cluster-api/blob/main/CONTRIBUTING.md#contributing-a-patch.

Reply:
Thanks, I get it.  I will open a new PR for it to unblock the other two commit.

Reply:
@qmloong you can 
keep `MachineHealthCheckSuccededCondition` (with the error) and mark it as deprecated/replaced by the new one
Add the new one (without the error) and use it in the codebase replacing all the usage of the old one

Reply:
OK, let me have try. Thank you.

Reply:
I open a new PR for this breaking change. 
#5861